src/components/todoitems/TodoItems.js

```diff
@@ -1,30 +1,20 @@
-import React, { memo } from 'react'
-import { useDispatch } from 'react-redux'
-import { completionStatus, moveToTrash } from '../../actions/todoActions'
+import React from 'react'
 import './todoitems.css'
 import { FaTrash } from 'react-icons/fa';
 
-const TodoItems = (props) => {
+export  const TodoItems = (props) => {
 
-    const dispatch = useDispatch(completionStatus())
-
-    const checkHandler = () => {
-        dispatch(completionStatus(props.data.id))
-    }
-
-    const trashhandler = () => {
-        dispatch(moveToTrash(props.data.id))
-    }
+    const { onCheckChange, onClick, data } = props
 
     return <div className='todoitem' >
         <div>
-            <input style={{ marginLeft: '15px', color: 'orange' }}
+            <input style={{ marginLeft: '15px',accentColor:'orange' }}
                 type='checkbox'
-                onChange={checkHandler} />
-            <label style={{ marginLeft: '20px' }}>{props.data.text}</label>
+                onChange={() => { onCheckChange(props) }} />
+            <label style={{ marginLeft: '20px' }}>{data.text}</label>
         </div>
-        <FaTrash style={{ marginRight: '5px', color: 'orange' }} onClick={trashhandler} />
+        <FaTrash style={{ marginRight: '5px', color: 'orange' }} onClick={() => { onClick(props) }} />
```

Comment:
pass only  required data in onClick function

Reply:
done